### PR TITLE
Stdout eda option

### PR DESCRIPTION
**Pull request by suppamax**
Following up #944 - a proposal for a new option for `eda` configuration, to notify that the tool outputs the result to STDOUT.

### CODE REVIEW

**Comment by suppamax** [comment]
I can take care of that, but unfortunately I cannot do anything before Easter.

**Comment by nmoroze** [comment]
Sounds good, thanks!

**Comment by nmoroze** [comment]
@suppamax thanks for updating this! I just wanted to double-check, is the updated version ready for review?

**Comment by suppamax** [comment]
It is.

File: siliconcompiler/core.py [hunk 1]
```diff
@@ -3775,14 +3775,43 @@ def read(fd):
                     import pty # Note: this import throws exception on Windows
                     retcode = pty.spawn(cmdlist, read)
             else:
-                with open(logfile, 'w') as log_writer, open(logfile, 'r') as log_reader:
+                stdoutFile = ''
```
Comment (nmoroze) [inline siliconcompiler/core.py:3778]: Use snake case for variable names, e.g. `stdout_file`.

File: siliconcompiler/core.py [hunk 2]
```diff
@@ -3775,14 +3775,43 @@ def read(fd):
...
+                if self.get('eda', tool, 'stdout', step, index, 'destination') == 'log':
+                    stdoutFile = step + "." + self.get('eda', tool, 'stdout', step, index, 'suffix')
```
Comment (nmoroze) [inline siliconcompiler/core.py:3780]: These lines are getting a little long, would recommend storing `suffix` in a temp variable to keep them shorter, e.g.:

```suggestion
                    suffix = self.get('eda', tool, 'stdout', step, index, 'suffix')
                    stdout_file = step + "." + suffix
```

File: siliconcompiler/core.py [hunk 3]
```diff
@@ -3775,14 +3775,43 @@ def read(fd):
...
+                    self._haltstep(step, index)
+
+                with open(stdoutFile, 'w') as stdout_writer, open(stdoutFile, 'r') as stdout_reader, open(stderrFile, 'w') as stderr_writer, open(stderrFile, 'r') as stderr_reader:
                     # Use separate reader/writer file objects as hack to display
                     # live output in non-blocking way, so we can monitor the
                     # timeout. Based on https://stackoverflow.com/a/18422264.
+                    isStdoutLog = self.get('eda', tool, 'stdout', step, index, 'destination') == 'log'
+                    isStderrLog = self.get('eda', tool, 'stderr', step, index, 'destination') == 'log' and stderrFile != stdoutFile
+                    if stderrFile == stdoutFile:
+                        stderrFile = subprocess.STDOUT
```
Comment (nmoroze) [inline siliconcompiler/core.py:3808]: What is the purpose of these lines?

**Review by nmoroze — CHANGES_REQUESTED** [review]
Great, this is looking awesome. We're getting closer! I left a few inline comments, mostly around style (e.g. renaming variables to be consistent with our `snake_case` convention).

Thanks!

File: siliconcompiler/schema.py [hunk 4]
```diff
@@ -1481,6 +1481,50 @@ def schema_eda(cfg, tool='default', step='default', index='default'):
...
+    scparam(cfg, ['eda', tool, 'stdout', step, index, 'destination'],
+            sctype='str',
+            defvalue='log',
+            scope='job',
+            shorthelp="Redirection for stdout",
+            switch="-eda_stdout_destination 'tool step index <str>'",
+            example=["cli: -eda_stdout_destination 'ghdl import 0 log'",
+                    "api: chip.set('eda','ghdl','stdout','import','0','destination','log')"],
+            schelp="""
+            Defines where to direct the output generated over stdout.""")
```
Comment (nmoroze) [inline siliconcompiler/schema.py:1493]: Would be nice to add one more sentence with list of supported options

**Comment by aolofsson** [comment]
This is a really nice addition! Thank you @suppamax for flagging the issue, the great implementation, and patience in working with us on the iterations!

**Pull request merged**